internal/db/errors.go

```diff
@@ -13,6 +12,10 @@ import (
 // https://www.postgresql.org/docs/11/errcodes-appendix.html.
 const pgUniqueViolation = "23505"
 
+type pgError interface {
```

Comment:
Why are we creating this? Is it just so we don't need pgconn?

Reply:
the updated gorm seems to return two different types of errors now.. this interface is here so that we don't need to care about the concrete type..we only ever call the one method on the two implementing types.